query/src/sql/planner/semantic/type_check.rs

```diff
@@ -553,6 +567,114 @@ impl<'a> TypeChecker<'a> {
         }
     }
 
+    pub async fn resolve_interval(
+        &mut self,
+        arg: &Expr<'a>,
+        interval_kind: &IntervalKind,
+        _required_type: Option<DataTypeImpl>,
+    ) -> Result<(Scalar, DataTypeImpl)> {
+        match interval_kind {
```

Comment:
Too long, the match should be inside the functions.

Reply:
seems can't do it. the function can't receive `IntervalKind`

Reply:
emm..seems can't do this. If we do this must, we will do some repeat work. I think it is unnecessary.